### DISCUSSION TITLE
Error compiling on MacOS

### DISCUSSION THREAD

**Discussion by chicodaman** [cvonk/OPNpool#12] [General]
Hi - I do a lot of ESP32 stuff using VS and PlatformIO and Arduino environments, but have never done a compile using ESP-IDF before, so I'm learning as I go. 

I get the following error when trying to compile on MacOS:

`Executing task: cmake -G Ninja -DPYTHON_DEPS_CHECKED=1 -DESP_PLATFORM=1 -B /Users/me/Documents/Arduino/OPNpool-master/build -S /Users/me/Documents/Arduino/OPNpool-master 

CMake Error: The source directory "/Users/me/Documents/Arduino/OPNpool-master" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.

 *  The terminal process "/bin/zsh '-c', 'cmake -G Ninja -DPYTHON_DEPS_CHECKED=1 -DESP_PLATFORM=1 -B /Users/me/Documents/Arduino/OPNpool-master/build -S /Users/me/Documents/Arduino/OPNpool-master'" failed to launch (exit code: 1).`

Any ideas what I'm doing wrong? 

(I have a real use for this code in my pool environment...thanks.)

**Comment by chicodaman** [comment]
Followup: Got MacOS to compile "blink" example but wouldn't compile OPNpool. Moved from MacOS to Windows, duplicated environment as described here, including downgrading esp-idf and vscode extension to versions mentioned in read.me. In all cases, both Windows and OSX terminate compilation with the same error:

OSX:
"-- Adding linker script /Users/me/esp/esp-idf/components/esp_rom/esp32/ld/esp32.rom.newlib-funcs.ld
-- Adding linker script /Users/me/esp/esp-idf/components/soc/esp32/ld/esp32.peripherals.ld
CMake Error at /Users/me/esp/esp-idf/tools/cmake/component.cmake:312 (message):
  Include directory
  '/Users/me/Documents/Arduino/ESP-IDF/OPNpool-master/interface/components/factory_reset_task/include'
  is not a directory."

Windows:
"-- Adding linker script C:/espressif/esp-idf/components/esp_rom/esp32/ld/esp32.rom.newlib-time.ld
-- Adding linker script C:/espressif/esp-idf/components/soc/esp32/ld/esp32.peripherals.ld
-- DEBUG: Use esp-modbus component folder: C:/espressif/esp-idf/components/freemodbus.
CMake Error at C:/espressif/esp-idf/tools/cmake/component.cmake:300 (message):
  Include directory
  'C:/MPLAB_Projects/ESP-IDF_Projects/OPNpool-master/interface/components/coredump_to_server/include'
  is not a directory."

Stuck...

**Comment by cvonk** [comment]
@chicodaman, 
Seems like you forgot the `--recursive` flag when cloning the project.  This flag causes required components to be downloaded, suck as `factory_reset_task` and `coredump_to_server`.

**Comment by chicodaman** [comment]
Thank you for the response. I was working on this as your message came in! I will try this, let you know how it goes. Trying to get it to compile on Windows first, then will move over to OSX.

**Comment by chicodaman** [comment]
Unfortunately, I'm having the same cloning issue mentioned by justlikeef in Issues..."Error trying to clone"  Issues: https://github.com/cvonk/OPNpool/issues/4

No solutions were provided in that thread. Any ideas how to resolve?

Thanks....

`C:\Users\me>git clone --recursive https://github.com/cvonk/OPNpool.git
Cloning into 'OPNpool'...
remote: Enumerating objects: 5171, done.
remote: Counting objects: 100% (89/89), done.
remote: Compressing objects: 100% (82/82), done.
remote: Total 5171 (delta 17), reused 54 (delta 6), pack-reused 5082
Receiving objects: 100% (5171/5171), 674.10 MiB | 6.37 MiB/s, done.
Resolving deltas: 100% (2966/2966), done.
Checking out files: 100% (271/271), done.
Submodule 'android' (git@github.com:cvonk/OPNpool_provisioning.git) registered for path 'android'
Submodule 'interface/components/coredump_to_server' (https://github.com/cvonk/ESP32_coredump-to-server) registered for path 'interface/components/coredump_to_server'
Submodule 'interface/components/factory_reset_task' (https://github.com/cvonk/ESP32_factory-reset-task.git) registered for path 'interface/components/factory_reset_task'
Submodule 'interface/components/ota_update_task' (https://github.com/cvonk/ESP32_ota-update-task.git) registered for path 'interface/components/ota_update_task'
Submodule 'interface/components/wifi_connect' (https://github.com/cvonk/ESP32_wifi-connect.git) registered for path 'interface/components/wifi_connect'
Cloning into 'C:/Users/me/OPNpool/android'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:cvonk/OPNpool_provisioning.git' into submodule path 'C:/Users/me/OPNpool/android' failed
Failed to clone 'android'. Retry scheduled
Cloning into 'C:/Users/me/OPNpool/interface/components/coredump_to_server'...
remote: Enumerating objects: 71, done.
remote: Counting objects: 100% (71/71), done.
remote: Compressing objects: 100% (44/44), done.
remote: Total 71 (delta 30), reused 54 (delta 19), pack-reused 0
Cloning into 'C:/Users/me/OPNpool/interface/components/factory_reset_task'...
remote: Enumerating objects: 46, done.
remote: Counting objects: 100% (46/46), done.
remote: Compressing objects: 100% (34/34), done.
remote: Total 46 (delta 14), reused 36 (delta 7), pack-reused 0
Cloning into 'C:/Users/me/OPNpool/interface/components/ota_update_task'...
remote: Enumerating objects: 82, done.
remote: Counting objects: 100% (82/82), done.
remote: Compressing objects: 100% (53/53), done.
remote: Total 82 (delta 38), reused 64 (delta 22), pack-reused 0
Cloning into 'C:/Users/me/OPNpool/interface/components/wifi_connect'...
remote: Enumerating objects: 119, done.
remote: Counting objects: 100% (119/119), done.
remote: Compressing objects: 100% (64/64), done.
remote: Total 119 (delta 58), reused 90 (delta 35), pack-reused 0
Receiving objects: 100% (119/119), 20.01 KiB | 6.67 MiB/s, done.
Resolving deltas: 100% (58/58), done.
Cloning into 'C:/Users/me/OPNpool/android'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
fatal: clone of 'git@github.com:cvonk/OPNpool_provisioning.git' into submodule path 'C:/Users/me/OPNpool/android' failed
Failed to clone 'android' a second time, aborting`

**Comment by chicodaman** [comment]
OK, fixed by adding an SSH key to my git account:  https://jhooq.com/github-permission-denied-publickey/

**Comment by cvonk** [comment]
The readme suggest to change to the interface directory before compiling.

**Reply by chicodaman** [reply]
Hi - I'm compiling successfully on both Windows and OSX. But I'm having issues connecting to wifi. 

I use the ESP-IDF SDK Configuration Editor to check "Use hardcoded Wi-Fi credentials, and enter my local SSID and password (I've tried both quoted and unquoted strings). I click Save, but it doesn't seem to actually save or write the updated config info, because the compiled ESP32 serial output says it's using SSID from nvram:

...
I (844) wifi_init: WiFi IRAM OP enabled
I (854) wifi_init: WiFi RX IRAM OP enabled
**W (854) main: Using SSID from nvram**
I (864) phy_init: phy_version 4670,719f9f6,Feb 18 2021,17:07:07
I (964) wifi:mode : sta (94:3c:c6:32:cd:10)
I (964) wifi:enable tsf

...and of course this occurs:

W (974) wifi_connect: WIFI_EVENT_STA_START
**W (3384) main: Wifi disconnect connectCnt=0, authErrCnt=0**
W (15784) main: Wifi disconnect connectCnt=0, authErrCnt=0 .
(repeats)...

So then I manually modify Kconfig as follows:

     config OPNPOOL_HARDCODED_WIFI_CREDENTIALS
        bool "Use hardcoded Wi-Fi credentials"
        default y
        help
            Use a hardcoded Wi-Fi credentials.  Set to n for BLE provisioning.
                    
    config OPNPOOL_HARDCODED_WIFI_SSID
        string "WiFi SSID"
        default "Linksys1200AC_2.4G"
        depends on OPNPOOL_HARDCODED_WIFI_CREDENTIALS
        help
            SSID (network name).
        
    config OPNPOOL_HARDCODED_WIFI_PASSWD
        string "WiFi Password"
        default "19500sweetwater"
        depends on OPNPOOL_HARDCODED_WIFI_CREDENTIALS
        help
            WiFi password (WPA or WPA2).

Now the ESP32 serial output says:

I (841) wifi_init: WiFi IRAM OP enabled
I (841) wifi_init: WiFi RX IRAM OP enabled
**W (861) main: Using SSID from Kconfig**
I (871) phy_init: phy_version 4670,719f9f6,Feb 18 2021,17:07:07
I (981) wifi:mode : sta (94:3c:c6:32:cd:10)
I (981) wifi:enable tsf

...but I still don't connect:

W (981) wifi_connect: WIFI_EVENT_STA_START
**W (3391) main: Wifi disconnect connectCnt=0, authErrCnt=0**
W (15791) main: Wifi disconnect connectCnt=0, authErrCnt=0
(repeats)...

If I hover over CONFIG_OPNPOOL_HARDCODED_WIFI_SSID in main.c, I can see that the value is correct:

![Screen Shot 2023-03-07 at 7 03 01 PM](https://user-images.githubusercontent.com/53490951/224189139-3e0a75cb-317f-4d5f-82c9-f9d7a12c09a1.jpg)

and likewise the password is revealed to be correct...but it doesn't connect.

Any ideas what I'm doing wrong here? 

Thanks...

**Reply by chicodaman** [reply]
Added some debug statements...

Indicates SSID and password are correct, but disconn->reason = "NO_AP_FOUND":

I (807) wifi_init: WiFi RX IRAM OP enabled
W (807) main: DBG: SSID is Sweetwater
W (807) main: DBG: SSID is 19500sweetwater
W (817) main: Using SSID from Kconfig
I (837) phy_init: phy_version 4670,719f9f6,Feb 18 2021,17:07:07
I (937) wifi:mode : sta (94:3c:c6:32:cd:10)
I (937) wifi:enable tsf
W (937) wifi_connect: WIFI_EVENT_STA_START
W (3347) wifi_connect: DBG: SSID is: Sweetwater
W (3347) wifi_connect: DBG: wifi password is: 19500sweetwater
W (3347) wifi_connect: DBG: Disconnect reason: NO_AP_FOUND
W (3357) main: Wifi disconnect connectCnt=0, authErrCnt=0
W (15757) wifi_connect: DBG: SSID is Sweetwater
W (15767) wifi_connect: DBG: SSID is 19500sweetwater
W (15767) wifi_connect: DBG: Disconnect reason: NO_AP_FOUND

hmmm....?

**Comment by chicodaman** [comment]
Interesting, though...I can get wifi to connect using the "test" directory of wifi-connect-master:

W (661) main: Using SSID from Kconfig
I (671) phy_init: phy_version 4670,719f9f6,Feb 18 2021,17:07:07
I (771) wifi:mode : sta (94:3c:c6:32:cd:10)
I (771) wifi:enable tsf
W (771) wifi_connect: WIFI_EVENT_STA_START
I (781) wifi:new:<1,0>, old:<1,0>, ap:<255,255>, sta:<1,0>, prof:1
I (1541) wifi:state: init -> auth (b0)
I (1551) wifi:state: auth -> assoc (0)
I (1581) wifi:state: assoc -> run (10)
I (1581) wifi:connected with Linksys1200AC_2.4G, aid = 38, channel 1, BW20, bssid = 14:91:82:bc:bb:0b
I (1591) wifi:security: WPA2-PSK, phy: bgn, rssi: -48
I (1611) wifi:pm start, type: 1

I (1611) wifi:AP's beacon interval = 102400 us, DTIM period = 1
I (2611) esp_netif_handlers: sta ip: 192.168.8.143, mask: 255.255.255.0, gw: 192.168.8.1
I (2611) main: WiFi connectCnt = 1
I (2611) main: Connected to WiFi AP
I (2611) main: Listening at http://192.168.8.143/

Not sure why the "interface" directory won't connect...

**Comment by cvonk** [comment]
Can you try the fix suggested in
  https://github.com/cvonk/OPNpool/issues/9

**Comment by chicodaman** [comment]
Yes! Quick cut and paste without detailed analysis (just before heading off to bed)...it connects and works.

Thanks much for the assistance. Looking forward to working with this, looks like a really nice implementation...
